ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/tabsontop/v1/tabsontop/clientlibs/site/js/tabs.js

```diff
@@ -326,10 +339,174 @@
+        }
+
+        addChild(childView) {
+            super.addChild(childView);
+            this.#cacheTemplateHTML(childView);
+        }
+
+        customRepeatableHtmlHandler(instanceManager,addedModel,htmlElement,beforeElement){
+            let result = this.#getBeforeViewElement(instanceManager, addedModel.index);
+            let beforeViewElement = result.beforeViewElement;
+            let elementToEnclose = result.elementToEnclose;
+            elementToEnclose.id=addedModel.id+result.idSuffix;
+            beforeViewElement.after(elementToEnclose);
+            elementToEnclose.append(htmlElement);
+        }
+
+
+        #cacheTemplateHTML(childView){
+            if(childView.getInstanceManager()!=null && (this._templateHTML==null || this._templateHTML[childView.getInstanceManager().getId()]==null)) {
+                var tabPanelToBeRepeated=childView.element.parentElement.parentElement;
+                var tabId=tabPanelToBeRepeated.id.substring(0,tabPanelToBeRepeated.id.lastIndexOf("__"))+"__tab";
+                var indexOfRepeatedTabPanel=this.#getTabIndexById(tabId)
+                var instanceManagerId = childView.getInstanceManager().getId();
+                var navigationTabToBeRepeated=this.#getCachedTabs()[indexOfRepeatedTabPanel];
+                var result=this.#getClosestFields(indexOfRepeatedTabPanel);
+                this._templateHTML[instanceManagerId] = {};
+                this._templateHTML[instanceManagerId]['navigationTab'] = navigationTabToBeRepeated;
+                this._templateHTML[instanceManagerId]['targetTabPanel'] = tabPanelToBeRepeated;
+                this._templateHTML[instanceManagerId]['closestNonRepeatableFieldId'] = result.closestNonRepeatableFieldId;
+                this._templateHTML[instanceManagerId]['closestRepeatableFieldInstanceManagerId'] = result.closestRepeatableFieldInstanceManagerId;
+            }
+        }
+
+
+
+        #getClosestFields(index){
+            var allTabPanels= this.#getCachedTabPanels();
+            var result={};
+            for (let i = 0; i < index; i++) {
+                var fieldId=allTabPanels[i].id.substring(0,allTabPanels[i].id.lastIndexOf("__"));
+                var fieldView=this.getChild(fieldId);
+                if(fieldView.getInstanceManager()==null){
+                    result["closestNonRepeatableFieldId"]=fieldId;
+                }else{
+                    result["closestRepeatableFieldInstanceManagerId"]=fieldView.getInstanceManager().getId();
+                }
+            }
+            return result;
+        }
+
+        #getTabIndexToInsert(closestNonRepeatableFieldId,closestRepeatableFieldInstanceManagerId){
+            var tabPanels=this.#getCachedTabPanels();
+            var closestNonRepeatableFieldFound=(closestNonRepeatableFieldId==null)?true:false;
+            var closestRepeatableFieldFound=(closestRepeatableFieldInstanceManagerId==null)?true:false;
+            var resultIndex=-1;
```

Comment:
pls leave spaces before and after operators for better readability.

Reply:
done, will add formatters in ide